src/components/timeline/TrackMemoryGraph.js

```diff
@@ -365,7 +373,9 @@ export const TrackMemoryGraph = explicitConnect<
     return {
       counter,
       threadIndex: counter.mainThreadIndex,
-      accumulatedSamples: counterSelectors.getAccumulateCounterSamples(state),
+      accumulatedSamples: counterSelectors.getAccumulateCounterSamples(
+        state
+      )[0],
```

Comment:
This selector returns an array now. But ideally this sample groups array should only be one element array. Before that code we check that already and if it's less than 1, we remove the counter all together. It's not possible this code to fail.

Reply:
I believe you should give the whole array to the component, just like we give the whole `counter` data, for more consistency.
Then the component code would do the `[0]` operation.

In the future, if we can have more sample groups, maybe the `sampleGroupIndex` will be passed as a ownProp ? (but let's not think about the future too much)